Cabal/Distribution/PackageDescription/Check.hs

```diff
@@ -200,15 +207,14 @@ checkSanity pkg =
         ++ "tool only supports up to version " ++ display cabalVersion ++ "."
   ]
   where
-    -- The public library gets special dispensation, because it
+    -- The public 'library' gets special dispensation, because it
     -- is common practice to export a library and name the executable
-    -- the same as the package.  We always put the public library
-    -- in the top-level directory in dist, so no conflicts either.
-    libNames = filter (/= unPackageName (packageName pkg)) . map libName $ libraries pkg
+    -- the same as the package.
+    subLibNames = catMaybes . map libName $ subLibraries pkg
```

Comment:
Mmm. I guess the invariant is that everything in `subLibraries` has a non-Nothing package name. I guess we can check that!